crates/red_knot_python_semantic/src/types.rs

```diff
@@ -377,15 +378,32 @@ impl<'db> FunctionType<'db> {
     /// annotated return type for this function, if any
     pub fn returns(&self, db: &'db dyn Db) -> Option<Type<'db>> {
         let definition = self.definition(db);
-        let DefinitionKind::Function(function_stmt_node) = definition.node(db) else {
-            panic!("Function type definition must have `DefinitionKind::Function`")
-        };
+        let function_stmt_node = Self::node(db, definition);
 
         function_stmt_node
             .returns
             .as_ref()
             .map(|returns| definition_expression_ty(db, definition, returns.as_ref()))
     }
+
+    pub fn params(&self, db: &'db dyn Db) -> impl Iterator<Item = Type<'db>> {
```

Comment:
Let's give this a doc comment to clarify its expected semantics.

I think the current semantics here are a little odd, given the name `params`, since it will iterate over all parameter annotations, skipping un-annotated parameters, and with no way to associate the iterated types with parameter names. I think at the very least this should probably iterate all parameters, and return pairs of `(param-node, annotated-type-or-unknown)`? Though I'm not sure that iterating all params is the right model here at all, given how this will be used. We may want a full-fledged `Signature` struct, that carries full information about all parameters and their types.

...I guess what I'm saying is maybe we'll need to design this API along with building out our initial use of it in checking call validity, to see what shape it should have?